### PR TITLE
CFY-7604: sudoers refactor

### PR DESCRIPTION
**Pull request by isaac-s**

### CODE REVIEW

File: cloudify_agent/shell/commands/configure.py [hunk 1]
```diff
@@ -42,24 +42,27 @@
               help='Indication whether sudo should be used when applying '
                    ' disable-requiretty part',
               is_flag=True)
+@click.option('--user',
+              help='(For disable-requiretty) The user for which to disable '
+                   'requiretty')
 @handle_failures
-def configure(disable_requiretty, relocated_env, no_sudo):
+def configure(disable_requiretty, relocated_env, no_sudo, user):
```
Comment (mcouthon) [inline cloudify_agent/shell/commands/configure.py]: If I remember correctly, this function is used both to disable TTY and to fix the venv, but you've changed only one invocation of it. Be sure to fix the other one as well.

File: cloudify_agent/resources/disable-requiretty.sh [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
+SUDOERS="/etc/sudoers"
+DISABLETTY_LINE="Defaults:${AGENT_USER} !requiretty"
+
+if [ -d "${SUDOERS_D}" ]; then
+    MY_SUDOERS=${SUDOERS_D}/cfy-${AGENT_USER}
+    echo "${SUDOERS_D} exists; adding ${MY_SUDOERS} to disable TTY requirement for ${AGENT_USER}"
+    ${MAYBE_SUDO} echo ${DISABLETTY_LINE} > ${MY_SUDOERS}
+elif [ -f "${SUDOERS}" ]; then
+    echo "${SUDOERS} exists; disabling TTY requirement for ${AGENT_USER}"
+    ${MAYBE_SUDO} echo ${DISABLETTY_LINE} >> ${SUDOERS}
```
Comment (mcouthon) [inline cloudify_agent/resources/disable-requiretty.sh]: Consider https://stackoverflow.com/questions/323957/how-do-i-edit-etc-sudoers-from-a-script

File: cloudify_agent/resources/disable-requiretty.sh [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
+DISABLETTY_LINE="Defaults:${AGENT_USER} !requiretty"
+
+if [ -z "${MAYBE_SUDO}" ]; then
+    MAYBE_SUDO="/bin/sh -c"
+fi
+
+if [ -d "${SUDOERS_D}" ]; then
+    MY_SUDOERS=${SUDOERS_D}/cfy-${AGENT_USER}
+    echo "${SUDOERS_D} exists; adding ${MY_SUDOERS} to disable TTY requirement for ${AGENT_USER}"
+    echo ${DISABLETTY_LINE} | ${MAYBE_SUDO} 'EDITOR="tee" visudo -f '${MY_SUDOERS}
```
Comment (mcouthon) [inline cloudify_agent/resources/disable-requiretty.sh]: It seems like there's no whitespace between the `'` and `${MY_SUDOERS}`. Is this by design?

Comment (mcouthon) [inline cloudify_agent/resources/disable-requiretty.sh]: Also, I didn't know `visudo` worked with anything other than `/etc/sudoers`. Neat!

**Comment by geokala** [comment]
Closed: As this PR is more than six months old without having been merged. Please reopen (and rebase if necessary) if it is still relevant.

File: cloudify_agent/shell/commands/configure.py (hunk 1 above)
Comment (isaac-s) [reply cloudify_agent/shell/commands/configure.py]: `--user` is optional (see `required=False`).

**Comment by mcouthon** [comment]
@isaac-s I kinda like to keep my slate clean on GitHub. Any chance to remove me from the reviewers here? (or close/merge the pr)?

**Comment by isaac-s** [comment]
@mcouthon sure, I'll unassign you

**Comment by isaac-s** [comment]
@mcouthon I'm unable to remove you — you'll have to remove yourself. I closed the PR.

**Comment by mcouthon** [comment]
@isaac-s that's fine, I'm good. Thank you both!

**Pull request closed without merging**